File: CS/Basic/recursion/Factorial.go
package main

import (
	"fmt"
)

func factorial(n int32) int64 {
	i_64 := int64(n)

	if i_64 <= 1 {
		return 1
	}

	return factorial(int32(i_64)-1) * i_64
}

func main() {
	var n int32
	fmt.Scan(&n)
	fmt.Println(factorial(n))
}

File: CS/Basic/recursion/Factorial.java
package CS.Basic.recursion;

public class Factorial {
    public static long factorial(int n) {
        if (n <= 1) {
            return 1;
        }

        return factorial(n - 1) * n;
    }

    public static void main(String[] args) {
        System.out.println(factorial(15));
    }
}
